### DISCUSSION TITLE
Tinkerpop Gremlin Support?

### DISCUSSION THREAD

**Discussion by th5** [kuzudb/kuzu#4172] [Ideas]
Kùzu looks like a great match for the Gremlin graph query language. We use it to express graph algorithms and queries on knowledge graphs. 

Is there interest in supporting Gremlin/Tinkerpop?

**Comment by semihsalihoglu-uw** [comment]
Hi Tom,

I do realize that Gremlin has its own community but I was never a fan of Gremlin as a standard API/language to interact with GDBMSs. That said, I do find value in extending our API to incrementally construct queries similar to how one constructs queries in Gremlin and akin to the ["Relational API" in DuckDB](https://duckdb.org/docs/api/python/relational_api). We have discussed implementing this at some point but not in our immediate [Phase 2 roadmap](https://github.com/kuzudb/kuzu/issues/981).

I am actually quite interested in understanding what you do with knowledge graphs and how (e.g., why you picked Gremlin to develop your algorithms). I could't find a link to your email or LinkedIn to ask you this separately but If you have some time and would like to meet with me/us, I'd be happy to set up a time to chat. You can contact me directly through email (which is in on my [website](https://cs.uwaterloo.ca/~ssalihog/)) or better by joining the [Kuzu slack](https://join.slack.com/t/kuzudb/shared_invite/zt-1qgxnn8ed-9LL7rfKozijOtvw5HyWDlQ).

Best,

Semih
